lib/asset_sync/storage.rb

```diff
@@ -243,8 +247,15 @@ def upload_file(f)
     end
 
     def upload_files
-      # get a fresh list of remote files
-      remote_files = ignore_existing_remote_files? ? [] : get_remote_files
+      remote_files = []
+      if self.cache_file && File.file?(self.cache_file)
+        log "start loading cache #{self.cache_file}"
+        remote_files = JSON.parse(File.read(self.cache_file))
```

Comment:
Do we have a failure path in case `JSON.parse` fail?
Fallback to `ignore_existing_remote_files? ? [] : get_remote_files`?

Reply:
Thanks!
When anything wrong happens, I think we should fall back to the current implementation with some warning message.